beater/beater.go

```diff
@@ -38,7 +40,22 @@ func (bt *beater) Run(b *beat.Beat) error {
 
 	go notifyListening(bt.config, pub.Send)
 
-	bt.server = newServer(bt.config, pub.Send)
+	settings := map[string]interface{}{}
+	if bt.config.Frontend.isEnabled() {
+		cfg := b.Config.Output.Config()
+		esClients, err := elasticsearch.NewElasticsearchClients(cfg)
+		if err != nil {
+			logp.Err(err.Error())
```

Comment:
I was thinking about it, as long as you are using ES for backend + Sourcemapping this shouldn't fail if you have configured it right. So yes for those cases we could panic.
The reason I only logged a message here is that the beats framework also allows for a configuration other than ES, eg. Kafka. If you are not using ES, then with this handling Sourcemapping simply doesn't work, but the server itself should still work. (Although I didn't test to configure something else than ES.)

Reply:
Do we plan to allow storing sourcemapping somewhere other than ES?  I understand if we allow providing sourcemaps through alternate means but if we're doing the mapping in apm server by fetching the sourcemap from ES, I think the ES client config has to be correct (but ES doesn't necessarily have to be available at startup time)

Reply:
Agree on that. 
However, right now the same pipeline is used for uploading `sourecemaps` as for uploading `transactions` or `errors`. If something else than ES is used, uploading sourcemaps would probably just fail right now. But it would also mean that no `output.hosts.elasticsearch` would be configured, which would lead to an error at this point. When panicing we would basically always fail to startup when someone configures a different output. 
When logging an error and continuing, as it is done here, we would just disable sourcemapping. I think this shouls be the way to go until we support sourcemapping with ES although another output is configured for `transactions` or `errors`.